tests/e2e/cucumber/steps/ui/shares.ts

```diff
@@ -145,3 +145,61 @@ When(
     await pageObject.navigate()
   }
 )
+
+When(
+  '{string} declines the following share(s)',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const shareObject = new objects.applicationFiles.Share({ page })
+
+    for (const resource of stepTable.hashes()) {
+      await shareObject.declineShare({ resource: resource.name })
+    }
+  }
+)
+
+When(
+  '{string} accepts the following share(s) from the context menu',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const shareObject = new objects.applicationFiles.Share({ page })
+    for (const resource of stepTable.hashes()) {
+      await shareObject.accept({ resource: resource.name, via: 'CONTEXT_MENU' })
+    }
+  }
+)
+
+When(
+  '{string} declines the following share from the context menu',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const shareObject = new objects.applicationFiles.Share({ page })
+
+    for (const resource of stepTable.hashes()) {
+      await shareObject.declineShare({ resource: resource.name, via: 'CONTEXT_MENU' })
+    }
+  }
+)
+
+When(
+  '{string} copies quick link of the resource {string} from the context menu',
+  async function (this: World, stepUser: string, resource: string): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const shareObject = new objects.applicationFiles.Share({ page })
+    await shareObject.copyQuickLink({
+      resource,
+      via: 'CONTEXT_MENU'
+    })
+  }
+)
+
+When(
+  '{string} should not be able to open the folder/file {string}',
+  async function (this: World, stepUser: string, resource: string): Promise<void> {
+    const { page } = this.actorsEnvironment.getActor({ key: stepUser })
+    const shareObject = new objects.applicationFiles.Resource({ page })
+    await shareObject.openFolder(resource).catch((err) => {
+      expect(err.message).toMatch(`Could not open resource ${resource}`)
+    })
```

Comment:
I am a little bit hesitant to asset the error message. Can we do something like the following? Determine the clickability or some state of the element or something else.
```suggestion
    expect(await <canIOpenFolder>).toBe(false)
```